src/Auth/Signup.tsx

```diff
@@ -1,118 +1,240 @@
-import React from "react";
+import React, { ChangeEvent, FormEvent, useState, useEffect } from "react";
 import "./Auth.css";
 import logo from "../images/logo2X.png";
 import google from "../images/google-logo.png";
 import kakao from "../images/kakao-logo.png";
 import { Link } from "react-router-dom";
+import useFormValidation from "./useFormValidation";
 
 const Signup = () => {
+  const {
+    errors,
+    validateEmail,
+    validateNickname,
+    validatePassword,
+    validatePasswordConfirm,
+  } = useFormValidation();
+  const [formValues, setFormValues] = useState({
+    email: "",
+    nickname: "",
+    password: "",
+    passwordConfirm: "",
+  });
+  const [isButtonActive, setIsButtonActive] = useState(false);
+  const [passwordVisible, setPasswordVisible] = useState(false);
+  const [passwordConfirmVisible, setPasswordConfirmVisible] = useState(false);
+
+  const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+    setFormValues({ ...formValues, [name]: value });
+  };
+
+  const handleBlur = (name: string, value: string) => {
+    switch (name) {
+      case "email":
+        validateEmail(value);
+        break;
+      case "nickname":
+        validateNickname(value);
+        break;
+      case "password":
+        validatePassword(value);
+        break;
+      case "passwordConfirm":
+        validatePasswordConfirm(formValues.password, value);
+        break;
+      default:
+        break;
+    }
+  };
+
+  const handleSubmit = (e: FormEvent) => {
+    e.preventDefault();
+    validateEmail(formValues.email);
+    validateNickname(formValues.nickname);
+    validatePassword(formValues.password);
+    validatePasswordConfirm(formValues.password, formValues.passwordConfirm);
+
+    if (
+      !errors.email &&
+      !errors.nickname &&
+      !errors.password &&
+      !errors.passwordConfirm
+    ) {
```

Comment:
signup 처럼 여러 필드에 대해 유효성검사가 필요한 경우 코드가 길고 복잡해지는 경우가 있는데요.
이 경우 every처럼 간략하게 작성할 수 있는 메서드를 활용해 보시면 좋을 것 같습니다!

```suggestion
    const isFormValid = Object.keys(errors).every((error) => !!error);

    if (isFormValid) {
```